mlrun/api/api/endpoints/projects.py

```diff
@@ -1,79 +1,67 @@
-@router.post("/project")
-def add_project(
-    project: schemas.ProjectCreate, db_session: Session = Depends(deps.get_db_session)
+@router.post("/projects", response_model=schemas.Project)
+def create_project(
+    project: schemas.Project, db_session: Session = Depends(deps.get_db_session)
 ):
-    project_id = get_db().add_project(db_session, project.dict())
-    return {
-        "id": project_id,
-        "name": project.name,
-    }
+    get_projects_manager().create_project(db_session, project)
+    return get_projects_manager().get_project(db_session, project.name)
 
 
 # curl -d '{"name": "p1", "description": "desc", "users": ["u1", "u2"]}' -X UPDATE http://localhost:8080/project
-@router.post("/project/{name}")
-def update_project(
-    project: schemas.ProjectUpdate,
+@router.put("/projects/{name}", response_model=schemas.Project)
+def store_project(
+    project: schemas.Project,
     name: str,
     db_session: Session = Depends(deps.get_db_session),
 ):
-    get_db().update_project(db_session, name, project.dict(exclude_unset=True))
-    return {}
+    get_projects_manager().store_project(db_session, name, project)
+    return get_projects_manager().get_project(db_session, name)
 
 
-# curl http://localhost:8080/project/<name>
-@router.get("/project/{name}", response_model=schemas.ProjectOut)
-def get_project(name: str, db_session: Session = Depends(deps.get_db_session)):
-    project = get_db().get_project(db_session, name)
-    if not project:
-        log_and_raise(error=f"project {name!r} not found")
+@router.patch("/projects/{name}", response_model=schemas.Project)
+def patch_project(
+    project: schemas.ProjectPatch,
+    name: str,
+    patch_mode: schemas.PatchMode = Header(
+        schemas.PatchMode.replace, alias="x-mlrun-patch-mode"
+    ),
+    db_session: Session = Depends(deps.get_db_session),
+):
+    get_projects_manager().patch_project(db_session, name, project, patch_mode)
+    return get_projects_manager().get_project(db_session, name)
 
-    project.users = [u.name for u in project.users]
 
-    return {
-        "project": project,
-    }
+# curl http://localhost:8080/project/<name>
+@router.get("/projects/{name}", response_model=schemas.Project)
+def get_project(name: str, db_session: Session = Depends(deps.get_db_session)):
+    return get_projects_manager().get_project(db_session, name)
 
 
 @router.delete("/projects/{name}", status_code=HTTPStatus.NO_CONTENT.value)
 def delete_project(
     name: str, db_session: Session = Depends(deps.get_db_session),
 ):
-    get_db().delete_project(db_session, name)
+    get_projects_manager().delete_project(db_session, name)
     return Response(status_code=HTTPStatus.NO_CONTENT.value)
 
 
 # curl http://localhost:8080/projects?full=true
-@router.get("/projects")
+@router.get("/projects", response_model=schemas.ProjectsOutput)
 def list_projects(
-    full: bool = False, db_session: Session = Depends(deps.get_db_session)
+    full: bool = True,
```

Comment:
format comes out a big ugly cause it's shadowing a built in name, so it's `format_` everywhere but ok

Reply:
yeah, no problem. Some other variants may also work, but it should be something that is indicative of a format change and not confuse with "we're going to return a partial list of elements/resources", and ideally also indicate what it's doing so it can be extended if other functionality is needed in the future